### ISSUE TITLE
DREAME L10s Pro Ultra Heat issue  with multiple segments.

### ISSUE THREAD

**Issue by vladeha** [Tasshack/dreame-vacuum#567]
I cannot add multiple rooms (segments) to a cleaning automation. Tried several configurations, but always only first segment cleaned, other segments ignored . Recommended configuration (that working with l20 s )doesn't work.
- service: dreame_vacuum.vacuum_clean_segment
 target:
 entity_id: vacuum.dreame
data:
 segments:
 - 1
 - 2
 - 3
repeats: 1

- Firmware Version 4.3.9_1527
- Integration Version  V.2.0.0.b.14

**Comment by vladeha** [comment]
Yes, actualy number of segmwents - 1,2 and 3 just examples. Robot can find all of them and clean all of them ...but only the first one - usually, in 90% of cases. So if I put segment 2 in first row, robot will clean it, if I put segment 3 in first row, robot will clean it ,etc.                                         
But sometimes, quite rarely, robot can clean all of them (I can see behaviour in phone app -if one segment marked (usually) or all of them (sometimes).

**Comment by vladeha** [comment]
Ok, I did it. How I can send you logs? What's your e-mail?  dreame-vacuum@noreply.github.com  ?

**Comment by paranerd** [comment]
Since I couldn't find it documented anywhere, I would like to use this thread to figure out how to connect the L10s Pro Ultra Heat to the dreame-vacuum integration in the first place.

AFAIK (and also tried) the robot doesn't connect with the Xiaomi Home App which of course causes the integration to not discover it.

How did you make this work?

**Comment by madkatz01** [comment]
> > Since I couldn't find it documented anywhere, I would like to use this thread to figure out how to connect the L10s Pro Ultra Heat to the dreame-vacuum integration in the first place.
> > AFAIK (and also tried) the robot doesn't connect with the Xiaomi Home App which of course causes the integration to not discover it.
> > How did you make this work?
> 
> Install the Beta version.

The beta is full. Any advice? I have the L10S Pro Ultra as well and can't connect it

**Comment by vladeha** [comment]
use dreame app

**Comment by Scorpion-HA** [comment]
I use it on my phone, but how can I make this integration / HomeAssistant to use my dreame app?

**Comment by paranerd** [comment]
For clarification: The beta version refers to this integration, NOT the Xiaomi/Dreame App.

**Comment by Scorpion-HA** [comment]
oh that helps a lot, thanks!
